packages/utils/vue3.3.polyfill.ts

```diff
@@ -0,0 +1,51 @@
+import type { Prop } from 'vue'
+// delete when upgrade to vue 3.3 : start
+// __ExtractPublicPropTypes copy from vue 3.3
+// If the type T accepts type "any", output type Y, otherwise output type N.
+// https://stackoverflow.com/questions/49927523/disallow-call-with-any/49928360#49928360
+type IfAny<T, Y, N> = 0 extends 1 & T ? Y : N
+
+type InferPropType<T, NullAsAny = true> = [T] extends [null]
+  ? NullAsAny extends true
+    ? any
+    : null
+  : [T] extends [{ type: null | true }]
+  ? any // As TS issue https://github.com/Microsoft/TypeScript/issues/14829 // somehow `ObjectConstructor` when inferred from { (): T } becomes `any` // `BooleanConstructor` when inferred from PropConstructor(with PropMethod) becomes `Boolean`
+  : [T] extends [ObjectConstructor | { type: ObjectConstructor }]
+  ? Record<string, any>
+  : [T] extends [BooleanConstructor | { type: BooleanConstructor }]
+  ? boolean
+  : [T] extends [DateConstructor | { type: DateConstructor }]
+  ? Date
+  : [T] extends [(infer U)[] | { type: (infer U)[] }]
+  ? U extends DateConstructor
+    ? Date | InferPropType<U, false>
+    : InferPropType<U, false>
+  : [T] extends [Prop<infer V, infer D>]
+  ? unknown extends V
+    ? keyof V extends never
+      ? IfAny<V, V, D>
+      : V
+    : V
+  : T
+
+type PublicRequiredKeys<T> = {
+  [K in keyof T]: T[K] extends { required: true } ? K : never
+}[keyof T]
+
+type PublicOptionalKeys<T> = Exclude<keyof T, PublicRequiredKeys<T>>
+
+declare module 'vue' {
```

Comment:
Perhaps a safer approach would be to remove the type declaration here and import the type directly using the file path, without having to modify `formattedText`.

Reply:
But this approach is not ideal for future upgrades. Currently, upgrading only requires replacing instances of __ExtractPublicPropTypes and changing some files. Using type imports would necessitate manually modifying numerous files.

Reply:
I don't think the difference is big. If upgrade in the future, you can just re-export the type here.
`export type { ExtractPublicPropTypes as __ExtractPublicPropTypes } from 'vue'`

Reply:
WDYT？ cc  @warmthsea @Dsaquel

Reply:
This will result in two import sources for the same type for a long time.

Reply:
Yes, this is not a good idea either.